### PR TITLE
Setup Jest for unit testing and add a few tests.

### LINKED ISSUE #12: Add some test
The current test step only does typescript check,
which is not enough to check for the overall integrity of the project.

Let's add some test cases to this repo that uses the utilities as well.

### PR DESCRIPTION
**Pull request by andrewlo**
fixes #12

### CODE REVIEW

File: test/math.test.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { clamp } from 'math';
+
+test('3 clamped by 1 & 2 should be 2', () => {
```
Comment (tzvipm) [inline test/math.test.ts]: these should all be in a `describe` block. Also, the test names should describe the particular feature, not the test case itself.

File: test/math.test.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import { clamp } from 'math';
```
Comment (tzvipm) [inline test/math.test.ts]: We generally prefer to use colocated tests, rather than a separate `src` and `tests` folder. Something like this:

```
src/
  math/
    tests/
      math.test.ts
    math.ts
    index.ts
  dom/
    tests/
      dom.test.ts
    dom.ts
    index.ts
  ...
```

The folder structure is a bit more verbose for creating utilities, but it is quite pleasant for working with existing utilities.

I wonder if we would like to keep that colocation structure in this case, though, since `javascript-utilities` is pretty flat, compared to repos like `polaris`.

@kaelig what are your thoughts?

File: circle.yml [hunk 3]
```diff
@@ -4,10 +4,12 @@ machine:
 
 dependencies:
   override:
-    - yarn
+    - yarn --ignore-engines
```
Comment (tzvipm) [inline circle.yml]: why is this needed?

File: package.json [hunk 4]
```diff
@@ -33,5 +36,22 @@
...
+    },
+    "testRegex": "(/__test__/.*|(\\.|/)(test|spec))\\.(js?|ts?)$",
+    "moduleDirectories": [
+      "node_modules",
+      "src"
+    ],
+    "moduleFileExtensions": [
+      "ts",
+      "tsx",
+      "js",
```
Comment (tzvipm) [inline package.json]: these shouldn't be needed; we only use `ts` in this repo for source files.

File: test/math.test.ts (hunk 2 above)
Comment (kaelig) [reply test/math.test.ts]: Thanks for asking ❤️ Agreed on colocating tests. 

On Polaris-styleguide we colocate tests at the same level as the modules themselves (without a “test” directory), I guess you could also do something like that: https://github.com/Shopify/polaris-styleguide/tree/master/server/styleguide-schema/content-parsers

**Comment by nimzco** [comment]
Closing since we shipped https://github.com/Shopify/javascript-utilities/pull/34 which adds jest.
